### PR TITLE
fix: Resolve account sequence errors and improve error handling. 

### LINKED ISSUE #22: account sequence mismatch
I am getting this error message very often during deployment. It seems there is some race condition between storing the code on terra and instantiating the contract

/usr/local/lib/node_modules/@iboss/terrain/node_modules/@oclif/errors/lib/index.js:26
        err = new cli_2.CLIError(input, options);
              ^

CLIError: account sequence mismatch, expected 478, got 477: incorrect account sequence
    at Object.error (/usr/local/lib/node_modules/@iboss/terrain/node_modules/@oclif/errors/lib/index.js:26:15)
    at Object.exports.instantiate (/usr/local/lib/node_modules/@iboss/terrain/lib/lib/deployment.js:69:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  oclif: { exit: 2 },
  code: undefined

### LINKED ISSUE #23: Account Mismatch Bug on Terrain?
**What I'm doing?**
I'm attempting to deploy a contract (specifically the counter.wasm from demo app) on the terra testnet via terrain.
```
terrain deploy counter --signer zx --network testnet
```

**What's going wrong?**
During each attempt at deployment I will receive an account mismatch sequence error. This sequence is always off by 1. And each time I do another deploy, **both** numbers get incremented by 1. I have attempted this about a dozen times.

```
CLIError: account sequence mismatch, expected 6, got 5: incorrect account sequence
```

**What I expect to happen?**
The contract to deploy to testnet and the response to contain the deployment address.

**What I've tried:**
I've tried the following (unsuccessfully):
- Generating a new wallet & keypair (and updating the key provider).
- Dropping and rebuilding the images and volumes for `cosmwasm/rust-optimizer`.

### PR DESCRIPTION
**Pull request by octalmage**
This pull request includes a couple of fixes and improvements. 

1. Manually incrementing the account sequence to avoid the dreaded account sequence mismatch error. Closes #22 Closes #23
2. Explicit timeout handling. In the case that a transaction does not get included in a block Terrain will now report this to the user. This can happen for a number of reason, more details on this [here](https://docs.terra.money/docs/develop/sdks/terra-js/common-examples.html#avoid-status-500-timed-out-waiting-for-tx-to-be-included-in-a-block).

```
storing wasm bytecode on chain... done
 ›   Error: transaction not included in a block before timeout
```
3. Check the response from broadcastSync to verify there aren't any obvious issues with the transaction, for example: 

```
storing wasm bytecode on chain... !
 ›   Error: 0uluna is smaller than 1000000uluna: insufficient funds: insufficient funds
```

### CODE REVIEW

File: src/lib/deployment.ts [hunk 1]
```diff
@@ -64,11 +65,35 @@ export const storeCode = async ({
...
 
-  const res = await lcd.tx.broadcast(storeCodeTx);
-  cli.action.stop();
+  const result = await lcd.tx.broadcastSync(storeCodeTx);
+  if (typeof result.code !== 'undefined') {
+    return cli.error(result.raw_log);
+  }
+
+  let res;
+  for (let i = 0; i <= 100; i++) {
```
Comment (zachalam) [inline src/lib/deployment.ts:74]: @octalmage - would recommend lowering the number of tries in this loop. If we can't get the txInfo by (perhaps 10/20 tries) it makes sense to stop and notify the end user.

Comment (octalmage) [reply src/lib/deployment.ts:74]: That’s fair! This is about a 30 second timeout currently, which does seem like a reasonable amount of time to retry. Maybe setting the sleep to 500ms, and only looping 50 times would be more appropriate?

**Comment by draflier** [comment]
When will this fix be implemented, I cannot deploy my contracts to testnet using terrain. :D

**Comment by ferencdg** [comment]
@iboss-ptk 

I realized this sequence number issue only happens(with the previous version of the code) when you connect to the 'official' bombay-12 nodes, if I setup my own testnode, it never happens. I suspect that the official bombay-12 node is just some kind of loadbalancer and forwards the request to different terra test nodes.

**Pull request merged**